pinecone/client.go

```diff
@@ -32,13 +33,10 @@ import (
 //
 // Fields:
 //   - Inference: An InferenceService object that exposes methods for interacting with the Pinecone [Inference API].
-//   - headers: An optional map of HTTP headers to include in each API request, provided through
-//     NewClientParams.Headers or NewClientBaseParams.Headers.
 //   - restClient: Optional underlying *http.Client object used to communicate with the Pinecone API,
 //     provided through NewClientParams.RestClient or NewClientBaseParams.RestClient. If not provided,
 //     a default client is created for you.
-//   - sourceTag: An optional string used to help Pinecone attribute API activity, provided through NewClientParams.SourceTag
-//     or NewClientBaseParams.SourceTag.
+//   - baseParams: A NewClientBaseParams object that holds the configuration for the Pinecone client.
```

Comment:
I think you might need to update the example code in this docstring to be `clientParams := pinecone.NewClientBaseParams{}` obj, no?

Reply:
You don't need to provide `pinecone.NewClientBaseParams{}`, you should be able to instantiate and interact with the client in the same way as before using `pinecone.NewClientParams{}`. I've just added a private field `baseParams` to bundle that up and reuse it when creating an index connection. There should be no visible impact to the user.

Reply:
Ahh okay thank you!